crates/pathfinder/src/sync/class_definitions.rs

```diff
@@ -139,6 +151,47 @@ pub(super) fn verify_layout(
     }
 }
 
+/// Checks if the streamed class is expected to be declared at the given block.
+/// If yes, it removes its hash from the "expected" set and returns the class, otherwise it returns an error.
+/// When the set of expected classes for the block is exhausted it fetches the set for the next block.
+///
+/// This function relies on the guarantee that the block numbers in the stream are correct.
+pub(super) async fn verify_declared_at(
+    storage: Storage,
+    expected_classes_for_block: Arc<Mutex<HashSet<ClassHash>>>,
```

Comment:
An alternative to consider: maybe use a `impl Stream<HashSet<ClassHash>>`? Then we spawn a task doing the storage thingy in a loop? Simplifies this code here by "moving" the db stuff outside which lets this function logic focus on only checking.